### PR TITLE
fixes #3059 ReplaceableWrapper does not work

### LINKED ISSUE #3059: ReplaceableWrapper does not work
In Plone 5.2.1 or coredev as of today on Python 3.8 if for some reason the code runs into Plone Portal Root `index_html` method the WSGI publisher can not render the result and raises:

```
Traceback (innermost last):
  Module ZPublisher.WSGIPublisher, line 156, in transaction_pubevents
  Module ZPublisher.WSGIPublisher, line 338, in publish_module
  Module ZPublisher.WSGIPublisher, line 256, in publish
  Module ZPublisher.mapply, line 84, in mapply
  Module ZPublisher.WSGIPublisher, line 62, in call_object
TypeError: 'ReplaceableWrapper' object is not callable
```

This happens to me in a customized site where I had no view set on the root. For some reason its not a problem with Python 2.7 AFAICT.

### PR DESCRIPTION
**Pull request by jensens**
See #3059 
Also deprecate ReplaceableWrapper

### CODE REVIEW

**Comment by jensens** [comment]
@jenkins-plone-org please run jobs

File: Products/CMFPlone/PloneFolder.py [hunk 1]
```diff
@@ -39,6 +41,10 @@ class ReplaceableWrapper:
     """A wrapper around an object to make it replaceable."""
 
     def __init__(self, ob):
+        warnings.warn(
+            'ReplaceableWrapper is deprecated. Planned remoaval in Plone 6.0',
```
Comment (thet) [inline Products/CMFPlone/PloneFolder.py]: nitpick: ``removal`` not ``remoaval``

Comment (thet) [reply Products/CMFPlone/PloneFolder.py]:
```suggestion
            'ReplaceableWrapper is deprecated. Planned removal in Plone 6.0',
```

Comment (thet) [inline Products/CMFPlone/PloneFolder.py]:
```suggestion
            'ReplaceableWrapper is deprecated. Planned removal in Plone 6.0',
```

**Pull request merged**